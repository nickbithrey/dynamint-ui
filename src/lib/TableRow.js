import React from 'react';
import PropTypes from 'prop-types';

const TableRow = ({record = []}) => (
	<tr>
		{record.map((r,i) => <td key={i}>{r}</td>)}
	</tr>
);

TableRow.propTypes = {
	record: PropTypes.arrayOf(PropTypes.node)
}

export default TableRow;