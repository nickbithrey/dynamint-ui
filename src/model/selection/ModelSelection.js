import React from 'react';
import { Link } from 'react-router-dom';
import SelectionComponent from '~/lib/SelectionComponent';

const initModel = {
	models: [
	    {col1: 'ref', col2: 'desc', col3: 'ACTIVE'},
	    {col1: 'ref2', col2: 'desc2', col3: 'ACTIVE'}
	]
}

export const ModelSelection = ({load, models = initModels}) => (
	<SelectionComponent 
		title={'Models'}
		list={models.models}
		loadList={load}
		columns={columns}
		createBtn={newBtn('/models/create', {}, 'Create')}
		editBtn={newBtn('/models/edit', {id: 1}, 'Edit')}
	/>
);

const columns = [
    {key: 'col1', name: 'Reference', fieldName: 'col1', minWidth: 50},
    {key: 'col2', name: 'Description', fieldName: 'col2', minWidth: 500},
    {key: 'col3', name: 'Status', fieldName: 'col3'}
];

const newBtn = (path, state, text) => {
	return {
		text: text,
		to: {
			pathname: path,
			state: state
		}
	};
}

export default ModelSelection;