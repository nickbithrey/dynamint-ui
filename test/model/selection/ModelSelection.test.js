import React from 'react';
import { shallow } from 'enzyme';

import ModelSelection from '~/model/selection/ModelSelection';
import TableRow from '~/lib/TableRow';

describe('<ModelSelection /> rendering', () => {

	it('renders standard record', () => {
		const details = {
			load: jest.fn(() => []), 
			models: {
				models: [
					{reference: 'reference', description: 'description', componentType: 'type'}
				]
			}
		};
		const wrapper = shallow( <ModelSelection {...details} /> );
		
		expect(wrapper).toMatchSnapshot();
	});
		
});