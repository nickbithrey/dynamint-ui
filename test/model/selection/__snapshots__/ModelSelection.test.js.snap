// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelSelection /> rendering renders standard record 1`] = `
<SelectionComponent
  columns={
    Array [
      Object {
        "fieldName": "reference",
        "key": "reference",
        "maxWidth": 150,
        "minWidth": 50,
        "name": "Reference",
      },
      Object {
        "fieldName": "componentType",
        "key": "componentType",
        "maxWidth": 50,
        "name": "Type",
      },
      Object {
        "fieldName": "description",
        "key": "description",
        "minWidth": 500,
        "name": "Description",
      },
    ]
  }
  createBtn={
    Object {
      "condition": [Function],
      "pathname": "/models/create",
      "stateFn": [Function],
      "text": "Create",
    }
  }
  editBtn={
    Object {
      "condition": [Function],
      "pathname": "/models/edit",
      "stateFn": [Function],
      "text": "Edit",
    }
  }
  list={
    Array [
      Object {
        "componentType": "type",
        "description": "description",
        "reference": "reference",
      },
    ]
  }
  loadList={[MockFunction]}
  title="Models"
/>
`;
